src/main/java/com/example/ajaxonboardingproject/security/CustomUserDetailsService.kt

```diff
@@ -11,7 +11,8 @@ import org.springframework.stereotype.Service
 class CustomUserDetailsService(private val userService: UserService) : UserDetailsService {
     @Suppress("SpreadOperator")
     override fun loadUserByUsername(username: String): UserDetails {
-        val user: User = userService.findByEmail(username)
+        val user: User = userService.findByEmail(username).block()
+            ?: throw NoSuchElementException("Can't get user with email: $username")

```

Comment:
```suggestion
        val user: User = userService.findByEmail(username)
                    .switchIfEmpty { NoSuchElementException("Can't get user with email: $username").toMono() }
                    .block()!!
            ?: throw NoSuchElementException("Can't get user with email: $username")
```

Note `.toMono()` - from separate lib: https://github.com/reactor/reactor-kotlin-extensions

As alternative to `switchIfEmpty` you can use `errorIfEmpty`(from the extension lib)

Reply:
Done!
Thanks for great extension (i've tried it)
But as i need only one method of it i preferred to use Mono.error()